src/app/qgsmaptooladdring.cpp

```diff
@@ -68,7 +69,8 @@ void QgsMapToolAddRing::polygonCaptured( const QgsCurvePolygon *polygon )
     return;
 
   vlayer->beginEditCommand( tr( "Ring added" ) );
-  const Qgis::GeometryOperationResult addRingReturnCode = vlayer->addRing( polygon->exteriorRing()->clone() );
+  QgsVectorLayerEditUtils utils( vlayer );
+  const Qgis::GeometryOperationResult addRingReturnCode = utils.addMultiRing( polygon->exteriorRing()->clone() );
```

Comment:
The selected features should be passed here (or handled later in `addMultiRing` implementation) or the map tool will ignore the actual selection and operate on all features.

Reply:
Indeed, I forgot to add the `selectedFeatureIds`!